### ISSUE TITLE
Allow the use of `..var` in i

### ISSUE THREAD

**Issue by markfairbanks** [Rdatatable/data.table#4958]
It would be useful if this was possible for use in `i`, much like it is possible to use in `j`:

``` r
library(data.table)

test_df <- data.table(x = 1:3, y = 1:3)
x <- 3

# Works in j
copy(test_df)[, new := y + ..x][]
#>    x y new
#> 1: 1 1   4
#> 2: 2 2   5
#> 3: 3 3   6

# Fails in i
test_df[y < ..x]
#> Error in .checkTypos(e, names_x): Object '..x' not found amongst x, y
```

**Comment by avimallu** [comment]
I suppose this can be now closed given the integration of #4304:
```r
> library(data.table)
> test_df <- data.table(x = 1:3, y = 1:3)
> x <- 3
> test_df[y < x, env = list(x = x)]
   x y
1: 1 1
2: 2 2
```

**Comment by markfairbanks** [comment]
I still think this would be worth adding even with the new addition. While the new `env` argument is useful, the `..var` syntax allows you to differentiate between the column `var` and the global `var`. For example:

``` r
library(data.table)

test_df <- data.table(x = 1:3, y = 1:3)
x <- 3

# Can refer to both column x and global x in j
copy(test_df)[, new := x + ..x][]
#>    x y new
#> 1: 1 1   4
#> 2: 2 2   5
#> 3: 3 3   6

# Can now only refer to global x
test_df[x < x, env = list(x = x)]
#> Empty data.table (0 rows and 2 cols): x,y
```

In this case it would be useful to be able to do:
```r
test_df[x < ..x]
```

**Comment by avimallu** [comment]
The `env` argument is the recommended method that replaces `get`, `mget` and `eval`, although not the `..` operator, as will be mentioned in the `datatable-programming` vignette when it is released:
https://github.com/Rdatatable/data.table/blob/79c3d3e1d2b1842edc948f2d8c895fe45a9070ac/vignettes/datatable-programming.Rmd#L369

**Comment by markfairbanks** [comment]
> although not the .. operator

If the `..` operator isn't being replaced, wouldn't that allow it to be expanded to work in i?

It's up to the data.table team of course, but I still think this functionality would be useful.

**Comment by jangorecki** [comment]
now you can have DIY prefixes
this should work as well
```r
test_df[x < ...x, env = list(...x = x)]
```

**Comment by dvg-p4** [comment]
I find `env = list(.x = x)` to be pretty ugly, and not a lot better than just 
```
.x = x
df[x == .x]
```

It would be very nice if `..` just worked out-of-the-box in `i` like it does in `j`.

**Comment by jangorecki** [comment]
I understand your point well.

New meta programming interface was developed having in mind it will be consumed by downstream packages and other software, like R scripts or shiny apps. Therefore more important than fast typing and reading was to be unambiguous (not being affected by scoping of overlapping names) and to be deeply customizable.

**Comment by dvg-p4** [comment]
True, you'd have the ambiguity right back if someone named a variable `..var` in the calling scope...but isn't that just as much a problem for `j`?

**Comment by jangorecki** [comment]
Yes it is

possibly you can use ..var in i already now, only not nested inside an expression

**Comment by dvg-p4** [comment]
Nope, doesn't seem to be recognized at all (unless that's changed in dev? I'm on 1.16.0)

**Comment by jangorecki** [comment]
No, then must not be there yet
